src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -35,20 +35,46 @@ class Lambda {
 
     val successfulPurges = CrierEventProcessor.process(distinctContentEvents) { event =>
       (event.itemType, event.eventType) match {
-        case (ItemType.Content, EventType.Delete) =>
-          sendFastlyPurgeRequestAndAmpPingRequest(event.payloadId, Hard, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(event.payloadId), config.fastlyDotcomApiKey)
+        case (ItemType.Content, EventType.Delete) => {
+          val dotcomCanonicalPurge = sendFastlyPurgeRequestAndAmpPingRequest(event.payloadId, Hard, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(event.payloadId), config.fastlyDotcomApiKey)
+          // Why is there no MAPI purge here?
+          val dotcomAliasPurges = extractAliasPaths(event).fold(Seq.empty[Boolean])({
+            _.map { aliasPath =>
+              sendFastlyPurgeRequestAndAmpPingRequest(aliasPath, Hard, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(aliasPath), config.fastlyDotcomApiKey)
+            }
+          })
+          dotcomCanonicalPurge && !dotcomAliasPurges.contains(false)
+        }
 
         case (ItemType.Content, EventType.Update) =>
           val contentType = extractUpdateContentType(event)
-          sendFastlyPurgeRequest(event.payloadId, Soft, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(event.payloadId), config.fastlyDotcomApiKey, contentType)
-          sendFastlyPurgeRequestForAjaxFile(event.payloadId, contentType)
-          sendFastlyPurgeRequest(event.payloadId, Soft, config.fastlyMapiServiceId, makeMapiSurrogateKey(event.payloadId), config.fastlyMapiApiKey, contentType)
+          val dotcomCanonicalPurge = sendFastlyPurgeRequest(event.payloadId, Soft, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(event.payloadId), config.fastlyDotcomApiKey, contentType)
+          val jsonPurge = sendFastlyPurgeRequestForAjaxFile(event.payloadId, contentType)
+          val mapiCanonicalPurge = sendFastlyPurgeRequest(event.payloadId, Soft, config.fastlyMapiServiceId, makeMapiSurrogateKey(event.payloadId), config.fastlyMapiApiKey, contentType)
+          val aliasPurges = extractAliasPaths(event).fold(Seq.empty[Boolean])({
+            _.map { aliasPath =>
+              val dotcomAliasPurge = sendFastlyPurgeRequest(aliasPath, Soft, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(aliasPath), config.fastlyDotcomApiKey, contentType)
```

Comment:
I- suspect- this repeat of the 3 send purge lines could be removed by resolving the event and alias paths into a seq(path,  ContentType) higher up... but not going to block on it and I'll probably be proven wrong if I start playing with the code =)

Reply:
I have been tempted to rethink this a couple of times too. I did start to refactor it last time but I think I got to the point where the duplication just moved elsewhere 🙄  Let me re-visit...